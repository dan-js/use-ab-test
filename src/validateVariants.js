export default (variants) =>
    variants.reduce((acc, variant) => acc + variant.percentage, 0) === 100
        ? true
        : (() => {
              throw new Error(
                  "Variant percentages should always add up to 100%"
              );
          })();
