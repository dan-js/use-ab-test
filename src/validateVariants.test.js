import validateVariants from "./validateVariants";

const invalidVariantsUnder = [
    {
        value: "hello",
        percentage: 10,
    },
    {
        value: "bonjour",
        percentage: 50,
    },
];

const invalidVariantsOver = [
    {
        value: "hello",
        percentage: 50,
    },
    {
        value: "bonjour",
        percentage: 55,
    },
];

const validVariants = [
    {
        value: "hello",
        percentage: 50,
    },
    {
        value: "hola",
        percentage: 50,
    },
];

describe("validateVariants()", () => {
    it("throws if variants do not add up to 100%", () => {
        expect(() =>
            validateVariants(invalidVariantsUnder)
        ).toThrowErrorMatchingSnapshot();
        expect(() =>
            validateVariants(invalidVariantsOver)
        ).toThrowErrorMatchingSnapshot();
    });

    it("returns true if variants do add up to 100%", () => {
        expect(validateVariants(validVariants)).toBe(true);
    });
});
