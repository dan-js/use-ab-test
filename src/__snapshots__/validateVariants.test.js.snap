// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateVariants() throws if variants do not add up to 100% 1`] = `"Variant percentages should always add up to 100%"`;

exports[`validateVariants() throws if variants do not add up to 100% 2`] = `"Variant percentages should always add up to 100%"`;
